lib/shelley/src/Cardano/Wallet/Shelley.hs

```diff
@@ -427,18 +400,86 @@ serveWallet
                 }
             )
             (neverFails "db layer should never forecast into the future"
-                $ timeInterpreter nl)
+                $ timeInterpreter netLayer)
             databaseDir
         Server.newApiLayer
-            walletEngineTracer params nl' tl db tokenMetaClient coworker
-      where
-        NetworkParameters gp sp pp = np
-        nl' = fromCardanoBlock gp <$> nl
+            walletEngineTracer
+            (block0, netParams, sTolerance)
+            (fromCardanoBlock genesisParameters <$> netLayer)
+            txLayer
+            dbFactory
+            tokenMetaClient
+            coworker
+
+withStakePoolLayer
+    :: Tracer IO StakePoolLog
+    -> Maybe Settings
+    -> DBLayer IO
+    -> NetworkParameters
+    -> NetworkLayer IO (CardanoBlock StandardCrypto)
+    -> ContT ExitCode IO StakePoolLayer
+withStakePoolLayer
+    tr
+    settings
+    dbLayer@DBLayer{..}
+    netParams@NetworkParameters{slottingParameters}
+    netLayer = lift $ do
+    gcStatus <- newTVarIO NotStarted
+    forM_ settings $ atomically . putSettings
+    void $ forkFinally
+        (monitorStakePools tr netParams netLayer dbLayer)
+        (traceAfterThread (contramap MsgExitMonitoring tr))
+
+    -- fixme: needs to be simplified as part of ADP-634
+    let startMetadataThread = forkIOWithUnmask
+            ($ monitorMetadata gcStatus tr slottingParameters dbLayer)
+    metadataThread <- newMVar =<< startMetadataThread
+    let restartMetadataThread = modifyMVar_ metadataThread $
+            killThread >=> const startMetadataThread
+    newStakePoolLayer gcStatus netLayer dbLayer restartMetadataThread
+
+withNetLayer
```

Comment:
```suggestion
withNetworkLayer
```

(likewise for other occurrences.) It's more verbose, but has the advantage that it matches the name of the type (`NetworkLayer`).

Reply:
I agree, but there is one unfortunate impediment: `withNetworkLayer` function already exists outside of this module (non `ContT`'d version) and it clashes with `withNetworkLayer` (`ContT`'d based). Do you have an idea on how to disambiguate them?